### PR TITLE
[orchagent]: Separate global SAI API pointers and initialization function

### PR DESCRIPTION
**Pull request by stcheng**
- To lighten the weight of main.cpp and make it easier for future refactor

### CODE REVIEW

File: orchagent/main.cpp [hunk 1]
```diff
@@ -13,42 +13,22 @@ extern "C" {
...
+#include "saiapi.h"
 
 using namespace std;
 using namespace swss;
 
 extern sai_switch_notification_t switch_notifications;
 
-#define UNREFERENCED_PARAMETER(P)       (P)
+extern sai_switch_api_t *sai_switch_api;
+extern sai_router_interface_api_t *sai_router_intfs_api;
```
Comment (qiluo-msft) [inline orchagent/main.cpp]: Move to the new header file

File: orchagent/saiapi.h [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#pragma once
+
+void initSaiApi();
```
Comment (qiluo-msft) [inline orchagent/saiapi.h]: Just feel the filenames confusing. How about saihelper.{h, cpp} ?

File: orchagent/main.cpp (hunk 1 above)
Comment (stcheng) [inline orchagent/main.cpp]: i want to keep them here as i want to have all api externs in the current *.cpp file for easy track, similar to all the other *.cpp files. Later, I could remove these externs with a SaiApi class to have all the pointers.

File: orchagent/saiapi.h (hunk 2 above)
Comment (stcheng) [inline orchagent/saiapi.h]: fixed.

File: orchagent/main.cpp (hunk 1 above)
Comment (qiluo-msft) [inline orchagent/main.cpp]: Could you elaborate 'easy track'? Actually prefer collect all the extern sai_xxx_api into one header file, which is more clean code.

Comment (stcheng) [inline orchagent/main.cpp]: easy to track which .cpp file uses what SAI API functions.

**Pull request merged**